suite-native/discovery/src/discoveryMiddleware.ts

```diff
@@ -45,16 +46,25 @@ export const prepareDiscoveryMiddleware = createMiddlewareWithExtraDeps(
         // to know the device state when starting discovery of newly authorized device.
         next(action);
 
-        // On successful authorization, create discovery instance and run it.
+        // On successful authorization, create discovery instance and run it with received device state.
         if (authorizeDeviceThunk.fulfilled.match(action) && isDeviceFirmwareVersionSupported) {
             dispatch(
                 startDescriptorPreloadedDiscoveryThunk({
-                    deviceState: action.payload.state,
                     areTestnetsEnabled,
+                    forcedDeviceState: action.payload.state,
                 }),
             );
         }
 
+        // this is for further continual rerunning discovery for various other reasons
+        if (
+            selectDeviceThunk.fulfilled.match(action) ||
```

Comment:
You need to check for `deviceActions.selectDevice` otherwise `discoveryCheckThunk` have still old state with previous device.

```suggestion
            deviceActions.selectDevice.match(action) ||
```

Reply:
77f34e6